dwave/optimization/src/nodes/indexing.cpp

```diff
@@ -212,7 +212,7 @@ struct AdvancedIndexingNodeData : NodeStateData {
 };
 
 struct AdvancedIndexingNode::IndexParser_ {
-    IndexParser_(Array* array_ptr, std::vector<array_or_slice>&& indices)
+    IndexParser_(Array* array_ptr, std::vector<array_or_slice>&& indices, ssize_t item_size)
```

Comment:
Why not get the `itemsize` from the array? Also, for consistency, I would use `itemsize` one word.